packages/shared/src/components/modals/LockedSquadModal.tsx

```diff
@@ -31,14 +32,21 @@ const options: PromptOptions = {
 };
 
 type LockedSquadModalProps = {
-  squad: Squad;
+  squadId: string;
 } & ModalProps;
+
 function LockedSquadModal({
-  squad,
+  squadId,
   onRequestClose,
 }: LockedSquadModalProps): ReactElement {
+  const { data: squad, isLoading } = useQuery(
+    [{ type: 'squad_locked', squadId }],
+    ({ queryKey: [{ squadId: id }] }) => getSquad(id),
+  );
   const { showPrompt } = usePrompt();
-  const [copying, copyLink] = useCopyLink(() => squad.permalink);
+  const token = squad?.currentMember?.referralToken ?? '';
+  const invitation = `${squad?.permalink}/${token}`;
```

Comment:
This would look weird in the case the currentMember is null right?
Is it even possible?

Reply:
Asking because you have the conditional after currentMember above.

Reply:
I am not fully familiar with how the `currentMember` is implemented at the moment, but being part of data that is coming from an API, I thought it could be nullable. If you think we can remove it, I'll refactor it.

Reply:
Yeah, it should always return something if you are part of the squad, based on logged in user.

Reply:
In the case we set the `initialData` for the query, we will have to keep the optional chaining since the passed `squad` from the props will have a null `currentMember`.

Reply:
Ah good point yes

Reply:
In that case, should we maybe set loading on the textfield?
That way the user know it's still busy

Reply:
I agree. Let's remove the empty fragment for loading and put it in the textfield now. Thank you for the suggestion! 🚢